helsinkioppii/templates/helsinkioppii/case_list_page.html

```diff
@@ -0,0 +1,77 @@
+{% extends "base.html" %}
+{% load i18n url_tags %}
+
+{% block body_class %}helsinkioppii-case-list-page{% endblock %}
+
+{% block content %}
+    <section class="filters">
+        <div class="container">
+            <form action="{{ page.url }}" method="get">
+                <div class="row">
+                    {% for field in form.visible_fields %}
+                        <div class="col-md-3 col-sm-6">
+                            {{ field.errors }}
+                            <label for="{{ field.id_for_label }}">
+                                {{ field.label }}
+                            </label>
+                            <div>
+                                {{ field }}
+                                {% if field.help_text %}
+                                    <p class="help">{{ field.help_text }}</p>
+                                {% endif %}
+                            </div>
+                        </div>
+                    {% endfor %}
+                </div>
+                <button type="submit" class="btn btn-primary">{% trans "Search" %}</button>
+            </form>
+        </div>
+    </section>
+
+    <section class="cases">
+        <div class="container">
+            {% for case in cases %}
+                <div class="case">
+                    <h3><a href="{{ case.url }}">{{ case.title }}</a></h3>
+                    <p class="abstract">{{ case.abstract }}</p>
+                </div>
+                {% if not forloop.last %}
+                    <hr>{% endif %}
+            {% endfor %}
+        </div>
+    </section>
+
+    <section class="case-pagination">
+        <div class="container">
+            {% if cases.has_other_pages %}
+                <ul class="pagination">
+                    {% if cases.has_previous %}
+                        <li>
+                            <a href="?{% update_get_parameters request "page" cases.previous_page_number %}">
```

Comment:
@ehaivala could you elaborate a bit what's happening here?
looks like you want to build URL for pagination here

Reply:
Yeah, I update the requests page GET parameter and preserve the other parameters. The parameters must be preserved or otherwise the applied filters would be undone when changing the page.